ocs_ci/utility/docker.py

```diff
@@ -0,0 +1,145 @@
+
+
+def get_latest_ds_olm_tag_docker(upgrade=False, latest_tag=None):
+    """
+    This function returns latest tag of OCS downstream registry or one before
+    latest if upgrade parameter is True
+
+    Args:
+        upgrade (str): If True then it returns one version of the build before
+            the latest.
+        latest_tag (str): Tag of the latest build. If not specified
+            config.DEPLOYMENT['default_latest_tag'] or 'latest' will be used.
+
+    Returns:
+        str: latest tag for downstream image from docker registry
+
+    Raises:
+        TagNotFoundException: In case no tag found
+
+    """
+    manifests = get_ocs_registry_docker_image_manifests()
+
+    latest_tag = latest_tag or config.DEPLOYMENT.get(
+        'default_latest_tag', 'latest'
+    )
+    latest_image = None
+    for manifest in manifests:
+        tag = manifest['tag']
+        image = manifest['fsLayers']
+        if tag == latest_tag:
+            latest_image = image
+            break
+    if not latest_image:
+        raise TagNotFoundException("Couldn't find latest tag!")
+    latest_tag_found = False
+    for manifest in manifests:
+        tag = manifest['tag']
+        image = manifest['fsLayers']
+        if not upgrade:
+            if (
+                tag not in constants.LATEST_TAGS
+                and image == latest_image
+            ):
+                return tag
+        if upgrade:
+            if not latest_tag_found and tag == latest_tag:
+                latest_tag_found = True
+                continue
+            if latest_tag_found:
+                if (
+                    tag not in constants.LATEST_TAGS
+                    and image != latest_image
+                    and "rc" in tag
+                ):
+                    return tag
+    raise TagNotFoundException("Couldn't find any desired tag!")
+
+
+def get_next_version_available_for_upgrade_docker(current_tag):
+    """
+    This function returns the tag built after the current_version
+
+    Args:
+        current_tag (str): Current build tag from which to search the next one
+            build tag.
+
+    Returns:
+        str: tag for downstream image from docker registry built after
+            the current_tag.
+
+    Raises:
+        TagNotFoundException: In case no tag suitable for upgrade found
+
+    """
+    if current_tag in constants.LATEST_TAGS:
+        return current_tag
+    manifests = get_ocs_registry_docker_image_manifests()
+    current_tag_index = None
+    for index, manifest in enumerate(manifests):
+        if manifest['tag'] == current_tag:
+            if index < 2:
+                raise TagNotFoundException("Couldn't find tag for upgrade!")
+            current_tag_index = index
+            break
+    sliced_reversed_tags = manifests[:current_tag_index]
+    sliced_reversed_tags.reverse()
+    for manifest in sliced_reversed_tags:
+        tag = manifest['tag']
+        if tag not in constants.LATEST_TAGS and "rc" in tag:
```

Comment:
we will need to fix this the same as I did in : https://github.com/red-hat-storage/ocs-ci/pull/1496/files but it's mostly related to upgrade so I will fix once we merge mentioned PR ^